core/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -188,6 +192,40 @@ class DeltaCatalog extends DelegatingCatalogExtension
     }
   }
 
+  override def loadTable(ident: Identifier, timestamp: Long): Table = {
+    loadTableWithTimetravel(ident, version = None, Some(timestamp))
+  }
+
+  override def loadTable(ident: Identifier, version: String): Table = {
+    loadTableWithTimetravel(ident, Some(version), timestamp = None)
+  }
+
+  /**
+   * Helper method which loads a Delta table with given time travel parameters.
+   * Only one of the timetravel inputs (version or timestamp) be present.
+   * @return
+   */
+  private def loadTableWithTimetravel(
+      ident: Identifier,
+      version: Option[String],
+      timestamp: Option[Long]): Table = {
+    val table = loadTable(ident)
+    assert((version.isDefined && timestamp.isEmpty) || (version.isEmpty && timestamp.isDefined),
+      "Expected the timetravel parameters consists of only one: either the timestamp or version")
+    table match {
+      case deltaTable: DeltaTableV2 =>
+        val ttOpts = Map(DeltaDataSource.TIME_TRAVEL_SOURCE_KEY -> "SQL") ++
+          (if (version.isDefined) Map(DeltaDataSource.TIME_TRAVEL_VERSION_KEY -> version.get)
+            else Map(DeltaDataSource.TIME_TRAVEL_TIMESTAMP_KEY ->
+              new Timestamp(timestamp.get / 1000).toString))
```

Comment:
It looks different from [DeltaTimeTravelSpec.parseTimestamp](https://github.com/delta-io/delta/blob/81d56794b1ea1032b1d7726c343325661e8ed20f/core/src/main/scala/org/apache/spark/sql/delta/DeltaTimeTravelSpec.scala#L129) that uses `DateTimeUtils.fromJavaTimestamp`. It begs unification.

Reply:
This is just a simple conversion of `microseconds` to `milliseconds`.

We will be using `DeltaTimeTravelSpec.parseTimestamp` later on, in a few function calls. `deltaTable.withOptions(ttOpts)` will call `DeltaDataSource.getTimeTravelVersion(options)` which will create `DeltaTimeTravelSpec`.